runtime/integration-tests/src/liquidity_pools/pallet/development/tests/liquidity_pools/foreign_investments.rs

```diff
@@ -1450,10 +1477,96 @@ mod mismatching_currencies {
 
 	use super::*;
 	use crate::{
-		liquidity_pools::pallet::development::{setup::CHARLIE, tests::register_usdt},
+		liquidity_pools::pallet::development::{
+			setup::CHARLIE,
+			tests::{
+				liquidity_pools::foreign_investments::setup::enable_usdt_trading, register_usdt,
+			},
+		},
 		utils::{GLMR_CURRENCY_ID, USDT_CURRENCY_ID},
 	};
 
+	#[test]
+	fn collect_for() {
+		TestNet::reset();
+		Development::execute_with(|| {
+			setup_pre_requirements();
+			let pool_id = DEFAULT_POOL_ID;
+			let investor: AccountId =
+				AccountConverter::<DevelopmentRuntime, LocationToAccountId>::convert((
+					DOMAIN_MOONBEAM,
+					BOB,
+				));
+			let pool_currency: CurrencyId = AUSD_CURRENCY_ID;
+			let foreign_currency: CurrencyId = USDT_CURRENCY_ID;
+			let pool_currency_decimals = currency_decimals::AUSD;
+			let invest_amount_pool_denominated: u128 = 6_000_000_000_000_000;
+			let sending_domain_locator = Domain::convert(DEFAULT_DOMAIN_ADDRESS_MOONBEAM.domain());
+			create_currency_pool(pool_id, pool_currency, pool_currency_decimals.into());
+			let invest_amount_foreign_denominated: u128 =
+				enable_usdt_trading(pool_currency, invest_amount_pool_denominated, true, || {});
+			do_initial_increase_investment(
```

Comment:
Nit but personally as a reader I like to have a line break between all the `lets` and an actual call 🙏 

```suggestion
			
			do_initial_increase_investment(
```